Assignments/Program.cs

```diff
@@ -1,3 +1,74 @@
-﻿using System;
+﻿// ---------------------------------------------------------------------------------------
+// Spark23 Assignments
+// Copyright (c) Metamation India.
+// ---------------------------------------------------------------------------------------
+// Program.cs
+// T16 - String Permutation
+// ---------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Spark;
+public class T17 {
+   public static void Main () {
+      Console.WriteLine ("\x1B[4m" + "String Permutator:-" + "\x1B[0m");
+      List<string> words = new () { "or", "not" };
+      foreach (string word in words) PrintResult (word);
+      Console.WriteLine ();
+      Console.Write ("Now, let's try entering a word: ");
+      PrintResult (Console.ReadLine ());
+   }
+
+   /// <summary> Prints the perumuted strings of given input word</summary>
+   public static void PrintResult (string input) {
+      Console.WriteLine ();
+      Console.WriteLine ($"Input word: {input}");
+      var answer = Permutated (input);
+      for (int i = 0; i < answer.Count; i++) {
+         Console.Write ($"{i + 1,2}. ");
+         Console.ForegroundColor = ConsoleColor.Green;
+         Console.WriteLine (answer[i]);
+         Console.ResetColor ();
+      }
+   }
+
+   /// <summary>Returns the list of permutated strings of given input word</summary>
+   public static List<string> Permutated (string input) {
+      input = input.ToUpper ();
+      if (input.Length == 2) return Permutated2 (input);
+      return Permutated34 (input);
+
+      /// <summary>Permutates given 2-letter word</summary>
+      static List<string> Permutated2 (string input)
+         => new () { $"{input[0]}{input[1]}", $"{input[1]}{input[0]}" };
+
+      /// <summary>Permutates given 3 [or] 4-letter word</summary>
+      static List<string> Permutated34 (string input) {

```

Comment:
Instead of Permuted2 Permuted34 etc. can you try doing this with recursive method to do this?